src/coreclr/jit/simdcodegenxarch.cpp

```diff
@@ -511,5 +511,46 @@ void CodeGen::genSimdUpperRestore(GenTreeIntrinsic* node)
     }
 }
 
+//-----------------------------------------------------------------------------
+// genSimd12UpperClear: Clears the upper 32-bits of a TYP_SIMD12 vector
+//
+// Arguments:
+//    tgtReg - The target register for which to clear the upper bits
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genSimd12UpperClear(regNumber tgtReg)
+{
+    assert(genIsValidFloatReg(tgtReg));
+
+    if (compiler->compOpportunisticallyDependsOn(InstructionSet_SSE41))
+    {
+        // ZMASK:   0b1000 - Preserve element 0, 1, and 2; Zero element 3
+        // COUNT_D: 0b11   - Insert into element 3
+        // COUNT_S: 0b11   - Insert from element 3
+
+        GetEmitter()->emitIns_SIMD_R_R_R_I(INS_insertps, EA_16BYTE, tgtReg, tgtReg, tgtReg, static_cast<int8_t>(0xF8));
+    }
+    else
+    {
+        // Preserve element 0, 1, and 2; Zero element 3
+
+        if (zroSimd12Elm3 == NO_FIELD_HANDLE)
+        {
+            simd16_t constValue;
+
+            constValue.u32[0] = 0xFFFFFFFF;
+            constValue.u32[1] = 0xFFFFFFFF;
+            constValue.u32[2] = 0xFFFFFFFF;
+            constValue.u32[3] = 0x00000000;
+
+            zroSimd12Elm3 = GetEmitter()->emitSimd16Const(constValue);
+        }
+
+        GetEmitter()->emitIns_SIMD_R_R_C(INS_addsd, EA_8BYTE, tgtReg, tgtReg, zroSimd12Elm3, 0);
```

Comment:
just curious - shouldn't it be packed AND?

Reply:
Yes, that's probably the cause of the jitstress failures I was just investigating 😆

Reply:
Fixed.